### ISSUE TITLE
lint:container command fails on named argument service

### ISSUE THREAD

**Issue by simonberger** [symfony/symfony#52768]
### Symfony version(s) affected

5.4.31

### Description

I receive an error like:
```
[ERROR] Invalid definition for service "service.one": argument 1 of
"ServiceOne::__construct()" accepts                    
         "Psr\Log\LoggerInterface", "null" passed.           
```

When executing
```
bin/console lint:container
```

In debugging I noticed it comes from `CheckTypeDeclarationsPass` where `::checkTypeDeclarations` expects a numbered list of arguments. I wonder why that just now fails as we have some more of these already, but anyway.

I checked the class in Symfony 6 and 7 and noticed @nicolas-grekas fixed that in PR https://github.com/symfony/symfony/pull/49944 which does sound like something different in general. Picking just the change of the for loop would fix the command for me.

### How to reproduce

Create a service with named arguments (not sure if this itself is really enough)

**Comment by xabbuh** [comment]
Can you create a small example application that allows to reproduce your issue?

**Comment by stof** [comment]
The fix done in https://github.com/symfony/symfony/pull/49944/files#diff-3fd13113f2559d9117d20f2727218989fbb6f762c20a0ae7edd372f6c32d4dcbR143 might indeed need to be backported to 5.4 (apparently, this was fixed in this PR because the PR itself was affected by the bug, without checking which branches were affected)

**Comment by simonberger** [comment]
I'll probably need (too) deep understanding of the value processing to create an example application. I already tried to debug why it runs into this just for this service and does not raise the issue for others.
I hoped nicolas who fixed it in another context and newer release or someone else is able to relate to it.

**Comment by simonberger** [comment]
It is a service with the following characteristics:
- named argument definition
- no autowiring. Enabling autowiring fixes it (skipping this function most likely).
- Leaving an argument with a default value out. Passing all arguments also fixes the problem

With this I should be able to create an example application tomorrow if still necessary.

**Comment by simonberger** [comment]
Here it is:
https://github.com/simonberger/symfony-gh52768-reproducer

```
composer install
bin/console lint:container
```

**Comment by xabbuh** [comment]
Thanks for the reproducer. 👍  Backporting the fix from #49944 indeed fixes this.
